File: backend-module/src/main/java/com/campus/backend/entity/Teacher.java
package com.campus.backend.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("teacher")
public class Teacher extends BaseEntity {
    private Long userId;
    private String teacherNo;
    private String name;
    private String title;
    private String subject;
    private String education;
    private String department;
    private Integer status;
}

File: frontend/src/types/teacher.ts
export interface Teacher {
  id: number
  userId: number
  teacherNo: string
  name: string
  title: string | null
  subject: string | null
  education: string | null
  department: string | null
  createdAt: string
  updatedAt: string
  isDeleted: boolean
}

export interface TeacherForm {
  id?: number
  userId?: number
  teacherNo: string
  name: string
  title: string | null
  subject: string | null
  education: string | null
  department: string | null
  username?: string
  password?: string
  phone?: string
}
